exchange/replication.go

```diff
@@ -333,27 +326,63 @@ func (r *Replication) handleRequest(s network.Stream) {
 	defer rs.Close()
 	req, err := rs.ReadRequest()
 	if err != nil {
+		log.Error().Err(fmt.Errorf("error when reading stream request : %v", err))
 		return
 	}
+
 	// Only the dispatch method is streamed directly at this time
 	switch req.Method {
 	case Dispatch:
 		// TODO: validate request
 		// Create a new store to receive our new blocks
 		// It will be automatically picked up in the TransportConfigurer
 		storeID := r.idx.ms.Next()
-		err = r.idx.SetRef(&DataRef{
-			PayloadCID:  req.PayloadCID,
-			PayloadSize: int64(req.Size),
-			StoreID:     storeID,
-		})
+
+		ref, err := r.idx.GetRef(req.PayloadCID)
```

Comment:
You don't need to get the ref here. It shouldn't exist at this point. Maybe you could use that as a way to prevent fetching twice if someone is spamming the network, then you'd need to return if the err isn't nil.